repoload.py

```diff
@@ -9,29 +9,57 @@
 """
 
 import argparse
+import configparser
 import subprocess
 import sys
 import json
 import os
 
 __VERSION__ = "0.1"
 
-# TODO Get this url from the manifest repo at the path '.repo/manifests.git/'.
-# This will mostly be the correct gerrit server url. Overwriting by a
-# enviornment variable should be only the fallback.
-# TODO if a environment variable is used, allow some sort of directory
-# selection.  Maybe the user has multiple repo checkouts from different gerrit
-# servers.
-try:
-    URL = os.environ['GERRIT_URL']
-except KeyError:
-    print("Set environment variable GERRIT_URL.", file=sys.stderr)
-    sys.exit(1)
-
 
 # Global Flags
 DEBUG = False
 
+MANIFEST_REPO_CFG = '.repo/manifests.git/config'
+
+
+def get_gerrit_url():
+    if 'GERRIT_URL' in os.environ:
+        return os.environ['GERRIT_URL']
+
+    # Search up in the tree to find the .repo directory.
+    # Usually, it is contained in the environment variable ANDROID_BUILD_TOP,
+    # but that might not be set.
+    cfg_path = ''
+    if 'ANDROID_BUILD_TOP' in os.environ:
+        cfg_path = '/'.join([os.environ['ANDROID_BUILD_TOP'],
+                             MANIFEST_REPO_CFG])
+    else:
+        d = os.getcwd()
+        cfg_path = '/'.join([d, MANIFEST_REPO_CFG])
+        found = os.path.isfile(cfg_path)
+        while not found and d != '/':
+            d = os.path.normpath('/'.join([d, '..']))
+            cfg_path = '/'.join([d, MANIFEST_REPO_CFG])
+            found = os.path.isfile(cfg_path)
+        if not found:
+            raise Exception('config file not found')
```

Comment:
I suggest to make this error message for descriptive, because it's the primary message a user sees, when (s)he is calling `repoload` in a non-repo source directory.

What about `Directory '.repo' not found in any parent directory. Not a repo checkout.`

Reply:
done